bench_test.go

```diff
@@ -0,0 +1,34 @@
+package smt
+
+import (
+	"crypto/sha256"
+	"strconv"
+	"testing"
+)
+
+func BenchmarkSparseMerkleTree_Update(b *testing.B) {
+	smn, smv := NewSimpleMap(), NewSimpleMap()
```

Comment:
Let's also invoke b.ReportAllocs() in all the benchmarks added in this PR.

Reply:
Ah right, I often call -benchmem explicitly from cli, that's why I don't often call b.ReportAllocs().